### PR TITLE
add on page load (#1883)

### LINKED ISSUE #1883: Add on page load method
### Description

The GUI might need a method to execute user method whenever a page is loaded (for data management, authorization, ...)

### Solution Proposed

_No response_

### Impact of Solution

_No response_

### Additional Context

_No response_

### Acceptance Criteria

- [ ] Ensure new code is unit tested, and check code coverage is at least 90%.
- [ ] Create related issue in taipy-doc for documentation and Release Notes.
- [ ] Check if a new demo could be provided based on this, or if legacy demos could be benefit from it.
- [ ] Ensure any change is well documented.

### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [X] I am willing to work on this issue (optional)

### PR DESCRIPTION
**Pull request by dinhlongviolin1**
Resolve #1883

### CODE REVIEW

**Review by FredLL-Avaiga — DISMISSED** [review]
👍

File: taipy/gui/gui.py [hunk 1]
```diff
@@ -143,6 +143,14 @@ class Gui:
             The signature of the *on_init* callback function must be:
 
             - *state*: the `State^` instance of the caller.
+        on_page_load (Callable): The function that is called when a page is loaded.<br/>
```
Comment (FabienLelaquais) [inline taipy/gui/gui.py]: I'm going to cry for more information.
"is loaded" is slightly vague, as we mentioned .

**Review by FabienLelaquais — DISMISSED** [review]
Perfect!

**Review by FabienLelaquais — APPROVED** [review]
👌

**Pull request merged**